cocotb/share/lib/gpi_log/gpi_logging.cpp

```diff
@@ -185,20 +185,23 @@ static void gpi_log_v(const char *name, enum gpi_log_levels level, const char *p
         goto error;
     }
 
-    PyObject *filter_ret = PyObject_CallFunctionObjArgs(pLogFilter, logger_name_arg, level_arg, NULL);
-    if (filter_ret == NULL) {
-        goto error;
-    }
+    {
+        // check if log level is enabled
+        PyObject *filter_ret = PyObject_CallFunctionObjArgs(pLogFilter, logger_name_arg, level_arg, NULL);
+        if (filter_ret == NULL) {
+            goto error;
+        }
 
-    int is_enabled = PyObject_IsTrue(filter_ret);
-    Py_DECREF(filter_ret);
-    if (is_enabled < 0) {
-        /* A python exception occured while converting `filter_ret` to bool */
-        goto error;
-    }
+        int is_enabled = PyObject_IsTrue(filter_ret);
+        Py_DECREF(filter_ret);
+        if (is_enabled < 0) {
+            /* A python exception occured while converting `filter_ret` to bool */
```

Comment:
```suggestion
            // A Python exception occured while converting `filter_ret` to bool
```

Reply:
Let's do that in a follow-up, it'll just make the diff harder to read.